File: CppPractice/CppLeetcode/Leetcode383.cpp
#inlcude<map>
class Solution {
public:
    bool canConstruct(string ransomNote, string magazine) {
        map<char,int> dictr;
        map<char,int> dictm;
        for(char c : ransomNote)
        {
            dictr[c]++;
        }
        for(char c : magazine)
        {
            dictm[c]++;
        }
        for(char c : ransomNote)
        {
            if (dictr[c]<dictm[c])
            {
                return false;
            }
        }
        return true;
    }
};

File: JavaScriptPractice/Leetcode383.js
var canConstruct = function(ransomNote, magazine) {
    let dictr={}
    let dictm={}
    for (let c of ransomNote) {
        if (dictr[c]) {
            dictr[c]++
        }
        else
        {
            dictr[c]=1
        }
    }
    for (let c of magazine) {
        if (dictm[c]) {
            dictm[c]++
        }
        else 
        {
            dictm[c]=1
        }
    }
    for (let c in dictr) {
        if (dictr[c]>(dictm[c]||0)) {
            return false
        }
    }
    return true
};